pylint/checkers/utils.py

```diff
@@ -1963,3 +1963,71 @@ def is_hashable(node: nodes.NodeNG) -> bool:
         return False
     except astroid.InferenceError:
         return True
+
+
+def _is_target_name_in_binop_side(
+    target: nodes.AssignName | nodes.AssignAttr, side: nodes.NodeNG | None
+) -> bool:
+    """Determine whether the target name-like node is referenced in the side node."""
+    if isinstance(side, nodes.Name):
+        if isinstance(target, nodes.AssignName):
+            return target.name == side.name  # type: ignore[no-any-return]
+        return False
+    if isinstance(side, nodes.Attribute):
```

Comment:
```suggestion
    if not isinstance(side, nodes.Attribute):
        return False
    if not isinstance(target, nodes.AssignAttr):
        return False
```
Then we can reduce everything 2 indentation level.

Reply:
Code has been changed as I created a new utility.